cluster/src/main/java/com/hartwig/pipeline/cram/CramConversion.java

```diff
@@ -0,0 +1,85 @@
+package com.hartwig.pipeline.cram;
+
+import com.hartwig.pipeline.Arguments;
+import com.hartwig.pipeline.ResultsDirectory;
+import com.hartwig.pipeline.alignment.AlignmentOutput;
+import com.hartwig.pipeline.execution.PipelineStatus;
+import com.hartwig.pipeline.execution.vm.BashCommand;
+import com.hartwig.pipeline.execution.vm.BashStartupScript;
+import com.hartwig.pipeline.execution.vm.ImmutableVirtualMachineJobDefinition;
+import com.hartwig.pipeline.execution.vm.InputDownload;
+import com.hartwig.pipeline.execution.vm.VirtualMachineJobDefinition;
+import com.hartwig.pipeline.execution.vm.VirtualMachinePerformanceProfile;
+import com.hartwig.pipeline.execution.vm.VmDirectories;
+import com.hartwig.pipeline.metadata.SingleSampleRunMetadata;
+import com.hartwig.pipeline.report.Folder;
+import com.hartwig.pipeline.report.RunLogComponent;
+import com.hartwig.pipeline.report.SingleFileComponent;
+import com.hartwig.pipeline.report.StartupScriptComponent;
+import com.hartwig.pipeline.stages.Stage;
+import com.hartwig.pipeline.storage.RuntimeBucket;
+
+import java.util.Collections;
+import java.util.List;
+
+public class CramConversion implements Stage<CramOutput, SingleSampleRunMetadata> {
+    public static final String NAMESPACE = "cram";
+    static final int NUMBER_OF_CORES = 6;
+
+    private final InputDownload bamDownload;
+
+    public CramConversion(final AlignmentOutput alignmentOutput) {
+        bamDownload = new InputDownload(alignmentOutput.finalBamLocation());
+    }
+
+    @Override
+    public List<BashCommand> inputs() {
+        return Collections.singletonList(bamDownload);
+    }
+
+    @Override
+    public String namespace() {
+        return NAMESPACE;
+    }
+
+    @Override
+    public List<BashCommand> commands(SingleSampleRunMetadata metadata) {
+        return new CramAndValidateCommands(bamDownload.getLocalTargetPath(),
+                VmDirectories.outputFile(CramOutput.cramFile(bamDownload.getLocalTargetPath()))).commands();
+    }
+
+    @Override
+    public VirtualMachineJobDefinition vmDefinition(BashStartupScript bash, ResultsDirectory resultsDirectory) {
+        return ImmutableVirtualMachineJobDefinition.builder()
+                .name("cram")
+                .startupCommand(bash)
+                .performanceProfile(VirtualMachinePerformanceProfile.custom(NUMBER_OF_CORES, 6))
+                .namespacedResults(resultsDirectory)
+                .build();
+    }
+
+    @Override
+    public CramOutput output(SingleSampleRunMetadata metadata, PipelineStatus jobStatus, RuntimeBucket bucket, ResultsDirectory resultsDirectory) {
+        String cram = CramOutput.cramFile(metadata.sampleName());
+        String crai = CramOutput.craiFile(metadata.sampleName());
+        Folder folder = Folder.from(metadata);
+        return CramOutput.builder()
+                .status(jobStatus)
+                .addReportComponents(
```

Comment:
I see we add the CRAM here to the final output I guess we want to remove the BAM from the aligner report components as well?

Reply:
Or maybe you were thinking we persist both to start?

Reply:
I was. But maybe that's overly cautious and not really beneficial.